src/actions/formActions.js

```diff
@@ -10,24 +12,36 @@ export const formActions = {
   TOGGLE_SHOW_FORM,
 };
 
-export const toggleShowForm = formName => ({
-  type: TOGGLE_SHOW_FORM,
-  formName,
-});
+export const addChip = (router, name, input, limit) => {
```

Comment:
It doesn't have to use react-router.  We can achieve the same effect directly with window.history.pushState.